packages/shared/src/components/Attachments/Attachments.tsx

```diff
@@ -92,19 +94,23 @@ export const Attachments = ({
     const onReplaceItemWithBrowse = (toReplace: Asset) => {
         setIsFlyoutOpen(false);
         appBridge.openAssetChooser(
-            (result: Asset[]) => {
-                onReplaceWithBrowse(toReplace, result[0]);
-                appBridge.closeAssetChooser();
+            async (result: Asset[]) => {
                 setIsFlyoutOpen(true);
+                appBridge.closeAssetChooser();
+                setAssetIdsLoading([...assetIdsLoading, toReplace.id]);
+                await onReplaceWithBrowse(toReplace, result[0]);
+                setAssetIdsLoading(assetIdsLoading.filter((id) => id !== toReplace.id));
             },
             {
                 multiSelection: false,
```

Comment:
```suggestion
                multiSelection: false,
                selectedValueIds: internalItems.map((item) => item.id),
```
add it also on line 89, for initial adds

to prevent that one asset could be added twice, which would end up in strange behaviours at dragging, etc...

Reply:
Thanks, fixed

Reply:
you missed to add it on line 107, otherwise its fine

Reply:
yep, fixed